src/layout/flex.rs

```diff
@@ -1,19 +1,183 @@
+/// Defines the options for layout flex justify content in a container.
 ///
-/// This controls how the space is distributed when the constraints are satisfied.
+/// This enumeration controls the distribution of space when layout constraints are met.
+///
+/// - `StretchLast`: Fills the available space within the container, putting excess space into the
+///   last element.
+/// - `Stretch`: Always fills the available space within the container.
+/// - `Start`: Aligns items to the start of the container.
+/// - `End`: Aligns items to the end of the container.
+/// - `Center`: Centers items within the container.
+/// - `SpaceBetween`: Adds excess space between each element.
+/// - `SpaceAround`: Adds excess space around each element.
 #[derive(Copy, Debug, Default, Display, EnumString, Clone, Eq, PartialEq, Hash)]
 pub enum Flex {
+    /// Fills the available space within the container, putting excess space into the last element.
+    ///
+    /// # Examples
+    ///
+    /// Length(40), Length(20)
+    ///
+    /// ```plain
+    /// 
+    /// Length(20), Length(10)
+    ///
+    /// <------------------------------------80 px------------------------------------->
+    ///
+    /// ┌──────────────────────────────────────┐┌──────────────────────────────────────┐
+    /// │                 40 px                ││                 40 px                │
+    /// └──────────────────────────────────────┘└──────────────────────────────────────┘
+    ///
+    /// Length(20), Fixed(10)
+    ///
+    /// <------------------------------------80 px------------------------------------->
+    ///
+    /// ┌────────────────────────────────────────────────────────────────────┐┌────────┐
+    /// │                                70 px                               ││  10 px │
+    /// └────────────────────────────────────────────────────────────────────┘└────────┘
+    /// ```
     #[default]
-    StretchLast, /* Always fill available space within the container putting excess into last
-                  * element */
-    Stretch,      // Always fill available space within the container
-    Start,        // Align items to the start of the container
-    End,          // Align items to the end of the container
-    Center,       // Center items within the container
-    SpaceBetween, // Add excess space between each element
-    SpaceAround,  // Add excess space around each element
+    StretchLast,
+    /// Always fills the available space within the container.
+    ///
+    /// # Examples
+    ///
+    /// ```plain
+    /// 
+    /// Length(20), Length(10)
+    ///
+    /// <------------------------------------80 px------------------------------------->
+    ///
+    /// ┌──────────────────┐┌──────────────────────────────────────────────────────────┐
+    /// │       20 px      ││                           60 px                          │
+    /// └──────────────────┘└──────────────────────────────────────────────────────────┘
```

Comment:
This is a weird result. Could do with a comment about why this is this way.

Is this comment flipped with StretchLast above?

Perhaps leave a line between the variants to communicate more obviously that the doc comment belongs to the variant below not the one above.

Reply:
Oops my bad. It should be the other way round.

Reply:
Fixed in latest commit.